Utilities/bootstrap

```diff
@@ -1012,6 +1032,11 @@ def main():
         build_flags.extend(["-Xswiftc", "-I{}".format(incdir)])
         build_flags.extend(["-Xswiftc", "-DHasCustomVersionString"])
     
+    if args.swift_build_dir:
+        swift_build_dir_search_path = os.path.join(args.swift_build_dir, "lib", "swift", args.sdk, args.arch)
+        build_flags.extend(["-Xlinker", "-L", "-Xlinker", swift_build_dir_search_path])
+        build_flags.extend(["-Xlinker", "-rpath-link", "-Xlinker", swift_build_dir_search_path])
```

Comment:
What is `-rpath-link`? I'm not familiar with that linker arg.

Reply:
@ddunbar, `-rpath-link` is an option that only considers the library path at link time, not at runtime.  No `DT_RPATH` entry is created in the emitted binary.

Reply:
I don't think macOS `ld64` accepts that option, so this code probably needs some more conditions in it?

Reply:
Yeah, I read it as being ELF-only, which does require some additional conditions it seems when re-reading this.